VK_App/VK_App/Controller/NewsViewController.swift

```diff
@@ -74,16 +88,37 @@ final class NewsViewController: UIViewController {
                 item.photoUrl = friend.userPhoto ?? ""
             }
         }
-        DispatchQueue.main.async {
-            self.news = news
-            self.tableView.reloadData()
-        }
+        return news
     }
 
     private func setupTableView() {
         tableView.dataSource = self
+        tableView.delegate = self
+        tableView.prefetchDataSource = self
         tableView.rowHeight = UITableView.automaticDimension
     }
+
+    @objc func refreshNewsAction() {
+        refreshControl.beginRefreshing()
+        let mostFreshNewsDate = news?.items.first?.date ?? Date().timeIntervalSince1970
+        networkService.fetchNews(startTime: mostFreshNewsDate + Constants.secondsDelay) { [weak self] result in
```

Comment:
```suggestion
        networkService.fetchNews(startTime: mostFreshNewsDate + Constants.secondsDelay) { [weak self] result in
```
это в отдельный метод лучше